R/getNeatOrder.R

```diff
@@ -0,0 +1,165 @@
+#' 
+#' # Transform the samples into relative abundances using CLR
+#' tse <- transformAssay(
+#'     tse, assay.type = "counts", method="clr", MARGIN = "samples",
+#'     name="clr", pseudocount = TRUE)
+#' 
+#' # Transform the features (taxa) into zero mean, unit variance
+#' # (standardize transformation)
+#' tse <- transformAssay(
+#'     tse, assay.type="clr", method="standardize", MARGIN = "features")
+#' 
+#' # Perform PCA using calculatePCA
+#' res <- calculatePCA(tse, assay.type = "standardize", ncomponents = 10)
+#' 
+#' # Sort by radial theta and sort the original assay data
+#' sorted_order <- getNeatOrder(res[, c(1,2)], centering = "mean")
+#' tse <- tse[, sorted_order]
+#' 
+#' # Define the color function and cap the colors at [-5, 5]
+#' col_fun <- colorRamp2(c(-5, 0, 5), c("blue", "white", "red"))
+#' 
+#' # Create the heatmap
+#' heatmap <- Heatmap(assay(tse, "standardize"),
+#'               name = "NeatMap",
+#'               col = col_fun,
+#'               cluster_rows = FALSE,  # Do not cluster rows
+#'               cluster_columns = FALSE,  # Do not cluster columns
+#'               show_row_dend = FALSE,
+#'               show_column_dend = FALSE,
+#'               row_names_gp = gpar(fontsize = 4), 
+#'               column_names_gp = gpar(fontsize = 6), 
+#'               heatmap_width = unit(20, "cm"),  
+#'               heatmap_height = unit(15, "cm")  
+#' )
+NULL
+
+#' @rdname getNeatOrder
+setGeneric("getNeatOrder", signature = c("x"),
+    function(x, centering = "mean", ...)
+        standardGeneric("getNeatOrder"))
+
+
+# Implementation for taking in a raw matrix.
+#' @rdname getNeatOrder
+#' @export
+setMethod("getNeatOrder", signature = c("matrix"),
+    function(x, centering = "mean", ...){
+        # Check args
+        .check_args(x, centering)
+        # Get the theta values and order them
+        theta_values <- .radial_theta(x, centering)
+        ordering <- .get_sorted_rownames(theta_values)
+        return(ordering)
+    }
+)
+
+# Checks the method arguments.
+.check_args <- function(x, centering) {
+    # Check data is a matrix
+    if (!is.matrix(x)) {
+        stop("Input data must be a matrix.", call. = FALSE)
+    }
+    # Check there is sufficient data
+    if (nrow(x) == 0 || ncol(x) == 0) {
+        stop(
+            "No data to plot. Matrix must have at least one row and one ",
+            "column.", call. = FALSE)
+    }
+    # Check there is sufficient data
+    if (ncol(x) != 2) {
+        stop("Matrix must have only 2 columns.", call. = FALSE)
+    }
+    # Check centering argument
+    centering <- match.arg(centering, c("mean", "median", "none"))
+    # Check for unique row names
+    if (any(duplicated(rownames(x)))) {
+        stop("Row names of the matrix must be unique.", call. = FALSE)
+    }
+    # Check for unique column names
+    if (any(duplicated(colnames(x)))) {
+        stop("Column names of the matrix must be unique.", call. = FALSE)
+    }
```

Comment:
Is unique column names required? I think they are not in this implementation.

Also can we work with index instead of rownames. That would make the function more robust. Can you check?

Reply:
That is what order() outputs

```
> order(c("A", 1, 4, 3, 5, 3))
[1] 2 4 6 3 5 1
```

Reply:
Otherwise, the function is good to go

Reply:
ok yeah thanks, just about to push and fix it will be a few mins

Reply:
Can you check if rownames are necessary?

Reply:
As in do you mean if we need the rowname uniqueness check? It works with both rownames and indices, and returns sorted row indices not rownames now.

Reply:
Yes, in practice TreeSE can have duplicated rownames. Can you chrck that the function works if there are no rownames or if they are duplicated

Reply:
Although matrix might require rownames, cannot remember by heart

Reply:
But it is good to make the function as general as possible. Usually it is better to work with row and column indices instead of names. Index is always unique.

Reply:
updated it now, yeah it works w no rownames and with duplicates